eng/common/pipelines/templates/steps/git-push-changes.yml

```diff
@@ -25,11 +25,22 @@ steps:
       echo "##vso[task.setvariable variable=HasChanges]$false"
       echo "No changes so skipping code push"
     }
+
+    # Conditions determined that SkipCheckingForChanges is 'false', set this
+    # as a variable which can be passed into the "Push Changes" step
+    echo "##vso[task.setvariable variable=SkipCheckingForChanges]false"
   displayName: Check for changes
   condition: and(succeeded(), eq(${{ parameters.SkipCheckingForChanges }}, false))
   workingDirectory: ${{ parameters.WorkingDirectory }}
   ignoreLASTEXITCODE: true
 
+- pwsh: |
```

Comment:
I'm not sure I'm following this logic. If `parameters.SkipCheckingForChanges` is true coming in then things will be correct as is what does this buy us?

Reply:
Ok I think I understand a little more based on your description in https://github.com/Azure/azure-sdk-tools/pull/3683.